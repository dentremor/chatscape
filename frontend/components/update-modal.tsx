import {
  Alert,
  AlertActionCloseButton,
  AlertActionLink,
} from "@patternfly/react-core";

export const UpdateModal = ({
  applyUpdate,
  dismissUpdate,
}: {
  applyUpdate: () => void;
  dismissUpdate: () => void;
}) => (
  <div className="pf-x-global-alerts-container pf-u-p-lg">
    <Alert
      variant="info"
      title="Update available"
      actionClose={<AlertActionCloseButton onClose={dismissUpdate} />}
      actionLinks={
        <>
          <AlertActionLink onClick={applyUpdate}>Update now</AlertActionLink>
          <AlertActionLink onClick={dismissUpdate}>Ignore</AlertActionLink>
        </>
      }
    >
      <p>A new version of ChatScape is available.</p>
    </Alert>
  </div>
);
