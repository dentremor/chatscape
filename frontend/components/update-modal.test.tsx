import { render } from "@testing-library/react";
import { UpdateModal } from "./update-modal";

describe("UpdateModal", () => {
  it("renders correctly", () => {
    expect(
      render(<UpdateModal applyUpdate={jest.fn()} dismissUpdate={jest.fn()} />)
    ).toMatchSnapshot();
  });
});
