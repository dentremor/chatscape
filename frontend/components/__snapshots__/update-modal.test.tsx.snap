// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateModal renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-x-global-alerts-container pf-u-p-lg"
      >
        <div
          aria-label="Info Alert"
          class="pf-c-alert pf-m-info"
          data-ouia-component-id="OUIA-Generated-Alert-info-1"
          data-ouia-component-type="PF4/Alert"
          data-ouia-safe="true"
        >
          <div
            class="pf-c-alert__icon"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              />
            </svg>
          </div>
          <h4
            class="pf-c-alert__title"
          >
            <span
              class="pf-u-screen-reader"
            >
              Info alert:
            </span>
            Update available
          </h4>
          <div
            class="pf-c-alert__action"
          >
            <button
              aria-disabled="false"
              aria-label="Close Info alert: alert: Update available"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
          </div>
          <div
            class="pf-c-alert__description"
          >
            <p>
              A new version of ChatScape is available.
            </p>
          </div>
          <div
            class="pf-c-alert__action-group"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-link pf-m-inline"
              data-ouia-component-id="OUIA-Generated-Button-link-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Update now
            </button>
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-link pf-m-inline"
              data-ouia-component-id="OUIA-Generated-Button-link-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Ignore
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-x-global-alerts-container pf-u-p-lg"
    >
      <div
        aria-label="Info Alert"
        class="pf-c-alert pf-m-info"
        data-ouia-component-id="OUIA-Generated-Alert-info-1"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Info alert:
          </span>
          Update available
        </h4>
        <div
          class="pf-c-alert__action"
        >
          <button
            aria-disabled="false"
            aria-label="Close Info alert: alert: Update available"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 352 512"
              width="1em"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
        </div>
        <div
          class="pf-c-alert__description"
        >
          <p>
            A new version of ChatScape is available.
          </p>
        </div>
        <div
          class="pf-c-alert__action-group"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Update now
          </button>
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Ignore
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
